custom-src/frontend/app/custom/kubernetes/list-types/kubernetes-nodes/kubernetes-nodes-list-config.service.ts

```diff
@@ -68,28 +76,79 @@ export class KubernetesNodesListConfigService implements IListConfig<KubernetesN
       cellFlex: '5',
     },
   ];
+  filters: IListFilter[] = [
+    {
+      default: true,
+      key: KubernetesNodesListFilterKeys.NAME,
+      label: 'Name',
+      placeholder: 'Filter by Name'
+    },
+    {
+      key: KubernetesNodesListFilterKeys.LABELS,
+      label: 'Labels',
+      placeholder: 'Filter by Labels'
+    },
+    {
+      key: KubernetesNodesListFilterKeys.IP_ADDRESS,
+      label: 'IP Address',
+      placeholder: 'Filter by IP Address'
+    }
+  ];
 
   pageSizeOptions = defaultHelmKubeListPageSize;
   viewType = ListViewTypes.TABLE_ONLY;
 
   enableTextFilter = true;
   text = {
-    filter: 'Filter by Name',
+    filter: '',
     noEntries: 'There are no nodes'
   };
 
   getGlobalActions = () => null;
   getMultiActions = () => [];
   getSingleActions = () => [];
   getColumns = () => this.columns;
-  getDataSource = () => this.nodesDataSource;
+  getDataSource = () => this.dataSource;
   getMultiFiltersConfigs = () => [];
+  getFilters = (): IListFilter[] => this.filters;
 
   constructor(
     store: Store<AppState>,
     kubeId: BaseKubeGuid,
   ) {
-    this.nodesDataSource = new KubernetesNodesDataSource(store, kubeId, this);
-  }
+    const transformEntities: DataFunction<KubernetesNode>[] = [
+      (entities: KubernetesNode[], paginationState: PaginationEntityState) => {
+        const filterKey = paginationState.clientPagination.filter.filterKey;
+        const filterString = paginationState.clientPagination.filter.string.toUpperCase();
+
+        switch (filterKey) {
+          case KubernetesNodesListFilterKeys.IP_ADDRESS:
+            return entities.filter(node => {
+              const internalIP: string = node.status.addresses.find(address => {
```

Comment:
It looks like its expected to only be a string, not a list of IPs, and documentation only references a single `InternalIP`
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.15/#nodeaddress-v1-core

It does look like it could be `ExternalIP` or `InternalIP` though so we could have a separate filter for either and handle it better if one isn't present.

Reply:
I've pushed a change that will fall back on the `ExternalIP` and also handle the case where neither are found. If the `ExternalIP` check goes against the spec of this feature or you'd rather it not there just let me know and will pull it out.